apps/aens/test/aens_utils_test.erl

```diff
@@ -0,0 +1,28 @@
+%%%=============================================================================
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%    Unit tests for AE Naming System utils
+%%% @end
+%%%=============================================================================
+
+-module(aens_utils_test).
+
+-include_lib("eunit/include/eunit.hrl").
+
+to_ascii_test() ->
+    ?assertEqual({error, no_registrar},
+                 aens_utils:to_ascii(<<"abcdefg">>)),
+    ?assertEqual({error, multiple_namespaces},
+                 aens_utils:to_ascii(<<"abcd.efgh.aet">>)),
+    ?assertEqual({error, registrar_unknown},
+                 aens_utils:to_ascii(<<"abcd.aettt">>)),
+    LongName = base64:encode(crypto:strong_rand_bytes(1000)),
+    ?assertEqual({error, name_too_long},
```

Comment:
I have not yet understood the significance of `aens_utils:to_ascii/1`, though I see some length checks here. I am concerned of:
* Off-by-one mistakes; and (orthogonally)
* Byte sequence of which to consider length (whole name? single label? if at least a Unicode char: the Unicode representation passed, a normalized Unicode representation of the string passed, the punycode representation of a normalized Unicode representation of the string passed?)

Some more tests may be beneficial (not sure whether here or in other points of the PR - still checking).

Reply:
Regarding length checks: this should be an option of Erlang-IDNA lib (shall be length of punycode representation), but pinned lib does not support it yet. Hence I added it here, to have very basic validation for now.

Reply:
Regarding length checks: this should be an option of Erlang-IDNA lib (shall be length of punycode representation), but pinned lib does not support it yet. Hence I added it here, to have very basic validation for now.